File: wlr/OGUR/OGUR/HUD/EquipmentHud.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Creatures;
using OGUR.Items;
using OGUR.Management;
using OGUR.Text;
using OGUR.Util;

namespace OGUR.HUD
{
    public class EquipmentHud:IHud
    {
        private Equipment m_equipment;
        private const string s_text = "Equipped";

        public EquipmentHud(ICreature owner, Equipment equipment): base(owner, XnaManager.WindowWidth / 2, XnaManager.WindowHeight / 2)
        {
            m_equipment = equipment;
        }

        public void Draw()
        {
            if (m_isVisible)
            {
                m_textHandler.Draw();
            }
        }

        private const string sep = ":";
        private const string newline = "\n";
        private string display = "EMPTY";
        private string title = "Equipped\n";

        public void Update(bool refresh)
        {
            if (m_isVisible)
            {
                m_textHandler.Update();
                m_textHandler.Clear();
                if (refresh)
                {
                    StringSquisher.Clear();
                    StringSquisher.Squish(title);
                    foreach (var item in m_equipment.GetItems())
                    {
                        StringSquisher.Squish(ItemSlot.Names[item.Key].Substring(0, 1), sep, item.Value.Name, newline);
                    }
                    display = StringSquisher.Flush();
                }
                m_textHandler.WriteDefault(display, 320, 30, GetHudOrigin());
            }
        }
    }
}

File: port-workspace/aigilas-ported/src/com/aigilas/hud/EquipmentHud.java
package com.aigilas.hud;import com.xna.wrapper.*;import java.util.*;import com.aigilas.creatures.*;import com.aigilas.items.*;import com.spx.util.*;import com.spx.core.*;
    public class EquipmentHud extends IHud
    {
        private Equipment _equipment;
        private static final String __text = "Equipped";

        public EquipmentHud(ICreature owner, Equipment equipment){ super(owner, XnaManager.WindowWidth / 2, XnaManager.WindowHeight / 2);            _equipment = equipment;
        }

        public void Draw()
        {
            if (_isVisible)
            {
                _textHandler.Draw();
            }
        }

        private static final String sep = ":";
        private static final String newline = "\n";
        private String display = "EMPTY";
        private String title = "Equipped\n";

        public void Update(boolean refresh)
        {
            if (_isVisible)
            {
                _textHandler.Update();
                _textHandler.Clear();
                if (refresh)
                {
                    StringSquisher.Clear();
                    StringSquisher.Squish(title);                    HashMap<Integer,GenericItem> items =_equipment.GetItems();
                    for (Integer item:items.keySet())
                    {
                        StringSquisher.Squish(ItemSlot.Names[item].substring(0, 1), sep, items.get(item).Name, newline);
                    }
                    display = StringSquisher.Flush();
                }
                _textHandler.WriteDefault(display, 320, 30, GetHudOrigin());
            }
        }
    }
